griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -119,7 +122,10 @@ def _base_params(self, prompt_stack: PromptStack) -> dict:
         # A max_tokens parameter is not required, but if it is specified by the caller, bound it to
         # the maximum value as determined by the tokenizer and pass it to the API.
         if self.max_tokens:
-            params["max_tokens"] = self.max_output_tokens(messages)
+            if isinstance(self.tokenizer, OpenAiTokenizer):
+                params["max_tokens"] = self.max_output_tokens(messages)
+            else:
+                params["max_tokens"] = self.max_output_tokens(self.prompt_stack_to_string(prompt_stack))
```

Comment:
Should this logic even exist? Seems like if a user specifies `max_tokens` we should pass that exact value. Otherwise we leave it empty and trust that the LLM provider will generate an appropriate amount of tokens.

Reply:
I think it works for OpenAI APIs but not for some others. In this particular case, we can probably drop it. I actually thought that @andrewfrench worked on this at some point to deal with GPT-4 rate limiting issues but I might be wrong.

Reply:
The work @andrewfrench did was to add the `if max_tokens` check. But agreed, I think we should remove the use of `max_output_tokens` in this case.

Reply:
Removed